prog/kube-utils/main.go

```diff
@@ -183,6 +190,42 @@ func reclaimPeer(weave weaveClient, cml *configMapAnnotations, peerName string,
 	return true, err
 }
 
+// regiesters with Kubernetes API server for node delete events. Node delete event handler
+// invokes reclaimRemovedPeers to remove it from IPAM so that IP space is reclaimed
+func registerForNodeUpdates(client *kubernetes.Clientset, stopCh <-chan struct{}, nodeName, peerName string) {
+	informerFactory := informers.NewSharedInformerFactory(client, 0)
+	nodeInformer := informerFactory.Core().V1().Nodes().Informer()
+	common.Log.Debugln("registering for updates for node delete events")
+	nodeInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		DeleteFunc: func(obj interface{}) {
+			node := obj.(*v1core.Node)
+			common.Log.Debugln("[kube-peers] Nodes deleted:", node.Name)
+			config, err := rest.InClusterConfig()
+			if err != nil {
+				common.Log.Fatalf("[kube-peers] Could not get cluster config: %v", err)
+			}
+			client, err := kubernetes.NewForConfig(config)
+			if err != nil {
+				common.Log.Fatalf("[kube-peers] Could not make Kubernetes connection: %v", err)
+			}
+			cml := newConfigMapAnnotations(configMapNamespace, configMapName, client)
+			weave := weaveapi.NewClient(os.Getenv("WEAVE_HTTP_ADDR"), common.Log)
```

Comment:
seems untidy to do this differently between `justReclaim` and `runReclaimDaemon` - this line could go inside `reclaimRemovedPeers`

Reply:
Agree. Moved the weave client creation part to `reclaimRemovedPeers`. To keep things tidy does it make sense to add `addMyselfToPeerList` to `runReclaimDaemon` as well (that the only difference i see between `justReclaim` and `runReclaimDaemon`?